### PR TITLE
Rename CourseRun to LearningResourceRun

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Preparation for #2239

#### What's this PR do?
- Renames `CourseRun` to `LearningResourceRun`.  
- Renames related classes like `CourseRunFactory`- > `LearningResourceRunFactory`
- Renames the attribute `course_run_id` to `run_id`

#### How should this be manually tested?
- Nothing should break, particularly:
  - Course home page
  - Course search
  - Course/bootcamp/program drawer
  - Importing mitx, xpro, ocw data
    ```
    get_xpro_data()
    sync_and_upload_edx_data()
    get_ocw_data()
   ```

### CODE REVIEW

File: course_catalog/api.py [hunk 1]
```diff
@@ -117,10 +117,10 @@ def parse_mitx_json_data(course_data, force_overwrite=False):
                     # we need to find the newest changes
                     max_modified = max(course_modified, course_run_modified)
 
-                    # Try and get the CourseRun instance. If it exists check to see if it needs updating
+                    # Try and get the LearningResourceRun instance. If it exists check to see if it needs updating
                     try:
-                        courserun_instance = course.course_runs.get(
-                            course_run_id=course_run.get("key")
+                        courserun_instance = course.runs.get(
```
Comment (aliceriot) [inline course_catalog/api.py]: should we rename this away from `courserun_instance`?

Comment (aliceriot) [reply course_catalog/api.py]: or maybe `run_instance`? idk, maybe it's not important

File: course_catalog/api.py [hunk 2]
```diff
@@ -234,14 +234,12 @@ def digest_ocw_course(
...
         course = ocw_serializer.save()
 
-        # Try and get the CourseRun instance.
+        # Try and get the LearningResourceRun instance.
         try:
-            courserun_instance = course.course_runs.get(
-                course_run_id=master_json.get("uid")
-            )
-        except CourseRun.DoesNotExist:
+            courserun_instance = course.runs.get(run_id=master_json.get("uid"))
```
Comment (aliceriot) [inline course_catalog/api.py]: wondering about this one too

File: course_catalog/etl/loaders.py [hunk 3]
```diff
@@ -57,17 +57,17 @@ def load_instructors(resource, instructors_data):
...
+    run_id = course_run_data.pop("run_id")
     instructors_data = course_run_data.pop("instructors", [])
     prices_data = course_run_data.pop("prices", [])
     topics_data = course_run_data.pop("topics", [])
 
-    course_run, _ = CourseRun.objects.update_or_create(
-        course_run_id=course_run_id,
-        object_id=course_or_bootcamp.id,
-        content_type=ContentType.objects.get_for_model(course_or_bootcamp),
+    course_run, _ = LearningResourceRun.objects.update_or_create(
```
Comment (aliceriot) [inline course_catalog/etl/loaders.py]: `course_run` -> `run` ?

File: course_catalog/etl/loaders_test.py [hunk 4]
```diff
@@ -205,35 +205,35 @@ def test_load_course(mock_upsert_tasks, course_exists, is_published):
 
 @pytest.mark.parametrize("course_run_exists", [True, False])
 def test_load_course_run(course_run_exists):
-    """Test that load_course_run loads the course run"""
-    course = CourseFactory.create(course_runs=None)
+    """Test that load_run loads the course run"""
+    course = CourseFactory.create(runs=None)
     course_run = (
```
Comment (aliceriot) [inline course_catalog/etl/loaders_test.py]: `run` ?

File: course_catalog/tasks.py [hunk 5]
```diff
@@ -170,7 +170,7 @@ def get_ocw_data(
             )
 
             # if course run synced before, update existing Course instance
-            courserun_instance = CourseRun.objects.filter(course_run_id=uid).first()
+            courserun_instance = LearningResourceRun.objects.filter(run_id=uid).first()
```
Comment (aliceriot) [inline course_catalog/tasks.py]: `run_instance`?

File: search/serializers_test.py [hunk 6]
```diff
@@ -218,20 +222,20 @@ def test_es_course_price_serializer():
 @pytest.mark.parametrize("has_full_name", [True, False])
 def test_es_course_run_serializer(has_full_name):
     """
-    Test that ESCourseRunSerializer correctly serializes a course run object
+    Test that ESRunSerializer correctly serializes a course run object
     """
     course_run = (
```
Comment (aliceriot) [inline search/serializers_test.py]: this one too?

File: search/serializers_test.py [hunk 7]
```diff
@@ -218,20 +222,20 @@ def test_es_course_price_serializer():
 @pytest.mark.parametrize("has_full_name", [True, False])
 def test_es_course_run_serializer(has_full_name):
```
Comment (aliceriot) [inline search/serializers_test.py]: should we rename this test too?

File: static/js/components/ExpandedLearningResourceDisplay.js [hunk 8]
```diff
@@ -46,8 +46,8 @@ const ExpandedLearningResourceDisplay = (props: Props) => {
   const courseRuns = [LR_TYPE_BOOTCAMP, LR_TYPE_COURSE].includes(
```
Comment (aliceriot) [inline static/js/components/ExpandedLearningResourceDisplay.js]: should we maybe just call this `runs` ?

File: static/js/lib/learning_resources.js [hunk 9]
```diff
@@ -163,7 +168,10 @@ export const minPrice = (prices: Array<CoursePrice>) => {
   return price > 0 && price !== Infinity ? `${formatPrice(price)}` : "Free"
 }
 
-export const getStartDate = (object: Object, courseRun: CourseRun) => {
+export const getStartDate = (
+  object: Object,
+  courseRun: LearningResourceRun
```
Comment (aliceriot) [inline static/js/lib/learning_resources.js]: rename this argument?

File: course_catalog/api.py (hunk 1 above)
Comment (mbertrand) [reply course_catalog/api.py]: It actually is a run for a course here, so I left it as is.

File: course_catalog/api.py (hunk 2 above)
Comment (mbertrand) [reply course_catalog/api.py]: Ditto - it's a run for an OCW course

File: course_catalog/etl/loaders.py (hunk 3 above)
Comment (mbertrand) [reply course_catalog/etl/loaders.py]: sure, it might be a program run here

File: course_catalog/etl/loaders_test.py (hunk 4 above)
Comment (mbertrand) [reply course_catalog/etl/loaders_test.py]: 👍

File: course_catalog/tasks.py (hunk 5 above)
Comment (mbertrand) [reply course_catalog/tasks.py]: Always a course run here

File: search/serializers_test.py (hunk 6 above)
Comment (mbertrand) [reply search/serializers_test.py]: 👍

File: search/serializers_test.py (hunk 7 above)
Comment (mbertrand) [reply search/serializers_test.py]: 👍

File: static/js/components/ExpandedLearningResourceDisplay.js (hunk 8 above)
Comment (mbertrand) [reply static/js/components/ExpandedLearningResourceDisplay.js]: 👍

File: static/js/lib/learning_resources.js (hunk 9 above)
Comment (mbertrand) [reply static/js/lib/learning_resources.js]: 👍

**Comment by mbertrand** [comment]
@aliceriot I needed to rebase again with master, let me know if you want to take another look at it before I merge.

**Pull request merged**